### ISSUE TITLE
Fix Serenity test reports

### ISSUE THREAD

**Issue by jasonaowen** [SolutionGuidance/psm#647]
Our integration tests produce JUnit test reports, but the Serenity Gradle plugin provides a `aggregate` target that collects the screenshots Serenity captures during the tests and builds some HTML around them. These Serenity test reports stopped working at some point - they now always say 0 tests.

Fix the Serenity test reports, so that `./gradlew integration-tests:test integration-tests:aggregate` lets us look at the screenshots.

Blocks #266 Publish Output from Serenity Tests.

**Comment by jasonaowen** [comment]
This may have been caused by #625, where we upgraded the Serenity libraries but not the Serenity plugin, due to conflicts with the JAXB Gradle plugin.

A useful first step, then, would be to build the JAXB classes, delete the JAXB parts of the build script and remove the JAXB Gradle plugin, and see if a newer version of the Serenity Gradle plugin fixes the problem. If so, then we can try to upgrade the JAXB Gradle plugin to see if a newer version of it is compatible with a newer version of the Serenity Gradle plugin; see also #545. If that doesn't work, we can try to find an alternate JAXB Gradle plugin, or downgrade Serenity.

Long term, of course, it would be nice to remove JAXB entirely (#596), but that's not feasible in the short term.

**Comment by jasonaowen** [comment]
Oh, and the JAXB plugin we're using appears to be no longer maintained, and has forked to https://github.com/rackerlabs/gradle-jaxb-plugin .

**Comment by cecilia-donnelly** [comment]
Took a stab at this, based on @jasonaowen's comment above.

1. Build the JAXB classes
    ```
    ./gradlew clean
    ./gradlew cms-portal-services:build
     [...]
    jaxb: starting Namespace Task
    jaxb: generating xsd namespace dependency tree
    ```

2. Deleted the JAXB parts of the build script (see https://github.com/SolutionGuidance/psm/commit/55b12851ace325c244ee5b7e2af56e16aa0aa52e for this and following)

3. Remove `jaxb-bindings.xjb`:

4. Remove jaxb reference in `XMLUtility.java`:

5. Remove the JAXB Gradle plugin (done in `build.gradle` change above).

6. Rebuild and deploy project:
```
cd psm-app
./gradlew cms-portal-services:build
cd ..
```

6. Update Serenity Gradle plugin from `net.serenity-bdd:serenity-gradle-plugin:1.4.0'` to `net.serenity-bdd:serenity-gradle-plugin:1.8.20'`

Success!

![screenshot from 2018-02-07 11-50-23](https://user-images.githubusercontent.com/1497818/35932646-d35a2d02-0bfd-11e8-9da4-84da203ae51a.png)

Looks like @jasonaowen was right about what's causing this.  Upgrades are the next step.

**Comment by slifty** [comment]
I updated to the latest jaxb over in 7b38844eeee6f129be22c71e386595d5f35eed28

Indeed serenity has a conflict here, which is actually introduced in the 1.5.x release. 

The error thrown is

```
...
Caused by: java.lang.NoClassDefFoundError: com/google/inject/internal/util/$Maps
        at com.google.inject.assistedinject.BindingCollector.<init>(BindingCollector.java:34)
        at com.google.inject.assistedinject.FactoryModuleBuilder.<init>(FactoryModuleBuilder.java:206)
        at org.openrepose.gradle.plugins.jaxb.schema.guice.DocSlurperModule.configure(DocSlurperModule.groovy:43)
...
```

I believe the issue is rooted in a mutual dependency between both Serenity and the JAXB plugin: Guice.  [Similar-looking errors](https://stackoverflow.com/questions/32228350/java-java-lang-noclassdeffounderror-com-google-inject-internal-util-precondi) in other projects out in the universe say that having mixed versions of Guice and a related library will trigger this.

[The offending line](https://github.com/rackerlabs/gradle-jaxb-plugin/blob/master/src/main/groovy/org/openrepose/gradle/plugins/jaxb/schema/guice/DocSlurperModule.groovy#L43) in the jaxb plugin comes from the assistedinject library (part of guice, but a separate dependency).

JAXB plugin and Serenity 1.4 both use Guice 3.0, but [Serenity 1.5 and above uses 4.x](https://github.com/serenity-bdd/serenity-core/compare/v1.4.1-rc.7...v1.5.1#diff-c197962302397baf3a4cc36463dce5eaL212).

I'm digging deeper to see if there's something we're doing wrong in the way we pull in the plugins.  It's worth noting that Gradle 4.6 changes the way plugins are pulled in, at least from a syntax perspective. I'm not suggesting that we move from Gradle 3.x to 4.x at this point, especially not until we confirm that the Guice dependency is the problem and understand why the conflict manifests.

**Comment by slifty** [comment]
Update: I'm going to take a break but I posted the question to StackOverflow for now: https://stackoverflow.com/questions/49101067/resolving-gradle-plugin-dependency-conflicts

**Comment by cecilia-donnelly** [comment]
@slifty fixed this in PR #694.  Thanks!
